### DISCUSSION TITLE
If i can convert dwg to shp by this?

### DISCUSSION THREAD

**Discussion by awefire** [NetTopologySuite/NetTopologySuite#488] [Q&A]
### Prerequisites

- [ ] I have written a descriptive issue title
- [ ] I have verified that I am running the latest version of NetTopologySuite
- [ ] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [ ] I have searched [open](https://github.com/NetTopologySuite/NetTopologySuite/issues) and [closed](https://github.com/NetTopologySuite/NetTopologySuite/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
If i can convert dwg to shp by this?If can,how to do?Can someone help me?
Thanks in advance.

**Comment by airbreather** [comment]
> convert dwg to shp

As of right now, there is no built-in support for reading .dwg files into GeoAPI / NetTopologySuite model objects.

If you can use your .dwg files to build GeoAPI / NetTopologySuite model objects in the first place, however, you could use the [NetTopologySuite.IO.ShapeFile](https://www.nuget.org/packages/NetTopologySuite.IO.ShapeFile/) package to write out the .shp files.

**Comment by DGuidi** [comment]
there was [this repo]( https://github.com/liberostelios/NetTopologySuite.IO.Dwg) too, no idea if is/was working or can be helpful

**Comment by awefire** [comment]
Thank you very much.But i'm not familiar with vb.Any other way?

**Comment by donnyv** [comment]
Your best bet is to do it in this order. 
dwg -> dxf -> read in dxf using [netDxf](https://github.com/haplokuon/netDxf) -> turn dxf into linestrings, etc  ->  NetTopologySuite.IO.ShapeFile
another way could be..
dwg -> dxf -> [use Ogr2ogr to convert to shp](https://gis.stackexchange.com/questions/66417/ogr2ogr-coverting-dxf-to-shapefile-why-am-i-losing-part-of-the-feature)

**Comment by DGuidi** [comment]
There's also this code that deserve a try, but I think should be heavily updated: https://github.com/AllanHao/NetTopologySuite.IO.Dwg-1

**Comment by donnyv** [comment]
> There's also this code that deserve a try, but I think should be heavily updated: https://github.com/AllanHao/NetTopologySuite.IO.Dwg-1

That library references two Autodesk libraries. So its not wholly contained.
